akka-http-core/src/main/scala/akka/http/impl/model/parser/SimpleHeaders.scala

```diff
@@ -95,7 +95,15 @@ private[parser] trait SimpleHeaders { this: Parser with CommonRules with CommonA
 
   // https://tools.ietf.org/html/rfc6265#section-4.2
   def `cookie` = rule {
-    oneOrMore(`cookie-pair`).separatedBy((ch(';') | ch(',')) ~ OWS) ~ EOI ~> (`Cookie`(_))
+    oneOrMore(`optional-cookie-pair`).separatedBy(';' ~ OWS) ~ EOI ~> { pairs ⇒
+      val validPairs = pairs.collect { case Some(p) ⇒ p }
+      `Cookie` {
+        if (validPairs.nonEmpty) validPairs
+        // Parsing infrastructure requires to return an HttpHeader value here but it is not possible
+        // to create a Cookie header without elements, so we create a dummy instance.
+        else HttpCookiePair("__dummy__", "__No_more_cookies_left_after_discarding_invalid_ones__") :: Nil
```

Comment:
I have dived a bit into how this piece works and I think I'd favour the throw a stacktrace less exception here actually - since there seems to be a place we can handle errors such as this already - inside the parseFull, so it'd be transparent to others?

If I understand the flow correctly (not sure due to one macro along the way) if we'd throw then we can handle a 
`case Left(e) if isSpecial(e)  ⇒ Right(RawHeader("", "")) // what header would it be...? again a special one on this layer hm :<` inside https://github.com/akka/akka/pull/18941/files?diff=split#diff-09a49db0c79cbd1ad03826f084fbd345R88 right?

I'm not clear about what RawHeader could be put there as a placeholder though, WDYM with that part of the proposal? Simply an "empty raw header"? (Related question - where is this __dummy__ filtered out, or we don't AFAICS (currently)?)

Do you think throwing would be performance wise acceptable? Should be measured I guess in any case..

// made me realise I've never used `DynamicRuleDispatch` so read up about it, interesting!